src/plugins/core/cell.ts

```diff
@@ -162,7 +162,7 @@ export class CellPlugin extends CorePlugin<CoreState> implements CoreState {
       for (let col = zone.left; col <= zone.right; col++) {
         for (let row = zone.top; row <= zone.bottom; row++) {
           const cell = this.getters.getCell({ sheetId, col, row });
-          if (cell) {
+          if (cell?.content) {
```

Comment:
note that `cell?.content` can also be quite costly if it's a formula. It rebuilds the entire formula string with the ranges.
```suggestion
          if (cell?.isFormula || cell?.content) {
```

Reply:
👍  completely forgot it was computed. I added the benchmark in the commit message but tl;dr  from 120ms -> 5ms